geopandas/io/tests/test_sql.py

```diff
@@ -181,3 +271,333 @@ def test_read_postgis_binary(self, connection_spatialite, df_nybb):
         )
         df = read_postgis(sql, con, geom_col=geom_col)
         validate_boro_df(df)
+
+    def test_write_postgis_default(self, df_nybb):
+        """Tests that GeoDataFrame can be written to PostGIS with defaults."""
+        engine = connect_engine("test_geopandas")
+        if engine is None:
+            raise pytest.skip()
+
+        table = "nybb"
+
+        # If table exists, delete it before trying to write with defaults
+        drop_table_if_exists(engine, table)
+
+        try:
+            # Write to db
+            write_postgis(df_nybb, con=engine, name=table, if_exists="fail")
+            # Validate
+            sql = "SELECT * FROM {table};".format(table=table)
+            df = read_postgis(sql, engine, geom_col="geometry")
+            validate_boro_df(df)
+        finally:
+            engine.dispose()
+
+    def test_write_postgis_fail_when_table_exists(self, df_nybb):
+        """
+        Tests that uploading the same table raises error when: if_replace='fail'.
+        """
+        engine = connect_engine("test_geopandas")
+        if engine is None:
+            raise pytest.skip()
+
+        table = "nybb"
+
+        try:
+            write_postgis(df_nybb, con=engine, name=table, if_exists="fail")
```

Comment:
Can you first write in once with `if_exists="replace"` to ensure it certainly exists? (otherwise it might depend on the order in which tests are run whether this is actually testing what we want to test)

Reply:
Done.